### PR TITLE
Modify comment timestamps and user votes to be stored in the database

### PR DESCRIPTION
**Pull request by IshaniSanturkar**
In this PR, I:
- modified code to add a new kind whose attributes are a user and a comment ID, representing all pairs of users and comments they have upvoted or downvoted. This is a more efficient way to prevent a user from upvoting or downvoting a comment multiple times rather than parsing and searching JSON each time. 
- modified the chart to pull data from timestamps stored in a different kind in the database so that on deleting a comment its timestamp entry would remain.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -132,18 +132,23 @@ private UserComment entityToComment(Entity entity) {
...
+    /*
+     * Has value:
+     * 1 if user has upvoted this comment
+     * -1 if user has downvoted this comment
+     * 0 if user has not voted for this comment
+     */
+    int voteValue = UtilityFunctions.getVoteInDatastore(userId, id);
+    if (voteValue != 0) {
+      votingStatus = (voteValue == 1) ?
+      UserComment.voteStatus.UPVOTED :
```
Comment (fenghaolw) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Do we need to distinguish UPVOTED and DOWNVOTED?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -475,6 +475,10 @@ function formatCommentReply(comment) {
  * isUpvote is false. 
  */
 function changeVote(comment, isUpvote, amount) {
+  // Prevent a POST request from changing vote count by more than 1
+  if(amount !== 1 && amount !== -1) {
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js]: space after `if`

and next line should be 2-space indentation

Cloud shell should be able to format JS file correctly?

**Review by fenghaolw — COMMENTED** [review]
you could probably try https://github.com/marketplace/actions/google-java-format and see if it can automatically format java code on commits?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (IshaniSanturkar) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The value is used in the client to determine which of the two buttons (if any) to highlight.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (IshaniSanturkar) [reply portfolio/src/main/webapp/script.js]: Sorry about that, I forgot to format the JS file. Cloud shell does automatically format JS files, but it keeps resetting the indentation to 4 space.

**Review by fenghaolw — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (fenghaolw) [reply portfolio/src/main/webapp/script.js]: You should be able to set the indentation to 2-space in cloud shell 
https://photos.app.goo.gl/qutP9pwXC9waV2a96

**Pull request merged**